File: bubblesort/bubblesort.py
#!/usr/bin/python

array = [2,2,1,3,4,5,6,300,24,7,11]

def bubbleSort(list):
    for i in range(0, len(list)):
        for j in range(1, len(list)-i):
            if list[j-1] > list[j]:
                tmp = list[j-1]
                list[j-1] = list[j]
                list[j] = tmp
    return list

def main():
    if __name__ == "__main__":
        print "before bubble sort:", array
        print "after bubble sort:", bubbleSort(array)

main()

File: bubblesort/bubblesort.go
package main

import (
	"fmt"
)

func main() {
	list := []int{2, 3, 4, 1, 10, 6, 7, 3, 3, 12}
	bubbleSort(list)
	fmt.Println(list)
}

func bubbleSort(list []int) {
	for i := 0; i < len(list); i++ {
		for j := 1; j < len(list)-i; j++ {
			if list[j-1] > list[j] {
				list[j-1], list[j] = list[j], list[j-1]
			}
		}
	}
}
